pkg/cloudmeta/gcp.go

```diff
@@ -35,16 +35,32 @@ func (gcp *gcpMetadata) Metadata(ctx context.Context) (InstanceMetadata, error)
 }
 
 func (gcp *gcpMetadata) getMachineType(ctx context.Context) (string, error) {
-	// The machine type for this VM. This value has the following format: projects/PROJECT_NUM/machineTypes/MACHINE_TYPE.
-	machineType, err := gcp.meta.GetWithContext(ctx, "instance/machine-type")
+	machineTypeResp, err := gcp.meta.GetWithContext(ctx, "instance/machine-type")
 	if err != nil {
 		return "", errors.Wrap(err, "gcp.meta.GetWithContext")
 	}
 
-	parts := strings.Split(machineType, "/")
-	if len(parts) < 2 {
-		return "", errors.Errorf("unexpected machine-type format: %s", machineType)
+	machineType, err := parseMachineTypeResponse(machineTypeResp)
+	if err != nil {
+		return "", err
+	}
+
+	return machineType, nil
+}
+
+// The machine type for this VM. This value has the following format: projects/PROJECT_NUM/machineTypes/MACHINE_TYPE.
+// See https://cloud.google.com/compute/docs/metadata/predefined-metadata-keys#instance-metadata.
+func parseMachineTypeResponse(resp string) (string, error) {
+	errUnexpectedFormat := errors.Errorf("unexpected machineType response format: %s", resp)
+
+	parts := strings.SplitN(resp, "/", 4)
+	if len(parts) != 4 {
+		return "", errUnexpectedFormat
+	}
```

Comment:
Why would you use `SplitN` instead of `Split`?

In a really really rare case of changed response syntax, `SplitN` only validates that the number of slashes is not smaller than 4, but it won't detect if the number of slashes is greater than 4.

And this is kind of picky, but from my POV using `SplitN` instead of regular `Split` should be done when there is some benefit behind it (as from my POV it's more "complicated" function variant), but it is arguably worse here.

Reply:
agree, Split would be better in that case! updated!